pkg/infrastructure/azure/loadbalancer.go

```diff
@@ -0,0 +1,238 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+)
+
+func createPublicIP(ctx context.Context, infraID, region string, pipClient *armnetwork.PublicIPAddressesClient) (*armnetwork.PublicIPAddress, error) {
+	publicIPAddressName := fmt.Sprintf("%s-pip-v4", infraID)
+	resourceGroupName := fmt.Sprintf("%s-rg", infraID)
```

Comment:
need to make all of these resourcegroupnames dynamic to account for user-provided values